src/imgui/imgui_raii_wrapper.h

```diff
@@ -0,0 +1,162 @@
+#ifndef MYPROJECT_IMGUI_RAII_WRAPPER_H
+#define MYPROJECT_IMGUI_RAII_WRAPPER_H
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <imgui.h>
+
+namespace irw {
+
+class Windowzita {
+public:
+  [[nodiscard]] explicit Windowzita(std::string const& id) : selected{ImGui::Begin(id.c_str())} {}
+  Windowzita(Windowzita const&) = delete;
+  Windowzita& operator=(Windowzita const&) = delete;
+  Windowzita(Windowzita&&) = delete;
+  Windowzita& operator=(Windowzita&&) = delete;
+  ~Windowzita() {
+    ImGui::End();
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabBar {
+public:
+  [[nodiscard]] explicit TabBar(std::string const& id) : selected{ImGui::BeginTabBar(id.c_str())} {}
+  TabBar(TabBar const&) = delete;
+  TabBar& operator=(TabBar const&) = delete;
+  TabBar(TabBar&&) = delete;
+  TabBar& operator=(TabBar&&) = delete;
+  ~TabBar() {
+    if (selected) {
+      ImGui::EndTabBar();
+    }
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabItem {
+public:
+  [[nodiscard]] explicit TabItem(std::string const& id) : selected{ImGui::BeginTabItem(id.c_str())} {}
+  TabItem(TabItem const&) = delete;
+  TabItem& operator=(TabItem const&) = delete;
+  TabItem(TabItem&&) = delete;
+  TabItem& operator=(TabItem&&) = delete;
+  ~TabItem() {
+    if (selected) {
+      ImGui::EndTabItem();
+    }
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class Text {
+public:
+  Text(char const* text_) : text{text_} {}
+  Text(std::string text_) : text{std::move(text_)} {}
+  Text(char const* text_, ImVec4 color_) : text{text_}, color{color_} {}
+  Text(std::string text_, ImVec4 color_) : text{std::move(text_)}, color{color_} {}
+  std::string get_text() const {
+    return text;
+  }
+  std::optional<ImVec4> get_color() const {
+    return color;
+  }
+  void write() const {
+    if (color) {
+      ImGui::TextColored(color.value(), "%s", text.c_str());
+    } else {
+      ImGui::Text("%s", text.c_str());
+    }
+  }
+private:
+  std::string text{};
+  std::optional<ImVec4> color{};
+};
+
+class Table {
+public:
+  using Row = std::vector<Text>;
```

Comment:
What is this?

Reply:
This is a type alias. I didn't want to write `std::vector<Text>` a bunch of times, so I aliased it. I also think it made the code clearer this way.